### PR TITLE
Dynamically set base URL in URL builder based on runtime

### PR DESCRIPTION
**Pull request by BenDMyers**
I noticed that folks are tripped up by the URL builder referencing a hardcoded `https://showmy.chat`. This leads to unintuitive behavior when validating changes locally.

This pull request sets up some global data leveraging environment variables (set locally and in Netlify's builds) to make the dev experience a bit more natural in this regard.

## Recommended validation

- Pull changes locally and run. Confirm the URL builder gives you a link with `https://localhost:8080` as its base
- Go the deploy preview at <https://deploy-preview-79--showmychat.netlify.app/>. Confirm that the URL builder gives you a link with the deploy preview's URL as its base.

### CODE REVIEW

**Review by NickyMeuleman — COMMENTED** [review]
Great, I've definitely ran into this myself, and this would make local development much easier!
Is using the `window` ok?
In that case: see my code-suggestion.

File: src/index.html [hunk 1]
```diff
@@ -123,7 +123,7 @@ <h1>Instant, themed Twitch chat overlays</h1>
 
 		// remove all parts of the url that are not query parameters
 		const channelName = values.get('channel') || defaultChannel;
-		let url = new URL(channelName, "https://showmy.chat/c/");
+		let url = new URL(channelName, "{{ site.baseUrl }}/c/");
```
Comment (NickyMeuleman) [inline src/index.html:126]:
```suggestion
		let url = new URL(channelName, `${window.location.origin}/c/`);
```
If accessing the `window` is ok, this single change has the same effect.
Then all the other code in this PR isn't needed. (and also works on local path other than localhost, like testing it on WSL)

Comment (BenDMyers) [reply src/index.html:126]: Oooh, excellent! I'll get right on that!

Comment (BenDMyers) [reply src/index.html]: Dang, yes, this is 100% the correct option.

**Pull request merged**